src/app/events/[id]/page.tsx

```diff
@@ -1,58 +1,40 @@
-import React from 'react';
-import Link from 'next/link';
+"use server";
 
-import PaymentsModal from '@/components/payments/PaymentsModal';
-import Button from '@/components/ui/Button';
-import * as StudentsTable from '@/components/ui/StudentsTable.client';
+import React from "react";
+import { cookies } from "next/headers";
 
-const backendUrl =
-  process.env.NEXT_PUBLIC_BACKEND_URL || 'http://localhost:3000';
+import { createRouteHandlerClient } from "@supabase/auth-helpers-nextjs";
+import {
+  dehydrate,
+  HydrationBoundary,
+  QueryClient,
+} from "@tanstack/react-query";
 
-const getEventData = async (id: string) => {
-  const response = await fetch(`${backendUrl}/event/${id}`);
+import ViewEventPage from "@/components/event/ViewEventPage";
+import { fetchEventData } from "@/utilities/fetch/event";
+import { fetchUser } from "@/utilities/fetch/user";
 
-  if (!response.ok) {
-    throw new Error('Failed to fetch event data!');
-  }
-
-  return response.json();
-};
-
-const getPaymentsData = async (id: string, paymentType: string) => {
-  const response = await fetch(`${backendUrl}/payment/${paymentType}`);
+const EventsPage = async ({ params }: { params: { id: string } }) => {
+  const queryClient = new QueryClient();
 
-  if (!response.ok) {
-    throw new Error('Failed to fetch event payments data!');
-  }
+  const cookieStore = cookies();
+  const supabase = createRouteHandlerClient({ cookies: () => cookieStore });
```

Comment:
pa change ko dri to `const supabase = createServerComponentClient({ cookies: () => cookieStore });`